summingbird-core/src/main/scala/com/twitter/summingbird/memory/Memory.scala

```diff
@@ -39,14 +41,14 @@ class Memory(implicit jobID: JobId = JobId("default.memory.jobId")) extends Plat
   type Plan[T] = Stream[T]
 
   private type Prod[T] = Producer[Memory, T]
-  private type JamfMap = HMap[Prod, Stream]
+  private type JamfMap = util.IdentityHashMap[Prod[_], Stream[_]]
```

Comment:
This is a good idea. I would really love to keep everything immutable too. Let me give this a shot.